rts/motoko-rts/src/types.rs

```diff
@@ -387,9 +445,16 @@ pub const TAG_ARRAY_SLICE_MIN: Tag = 32;
 #[repr(C)] // See the note at the beginning of this module
 pub struct Obj {
     pub tag: Tag,
+    /// Forwarding pointer to support object moving in the incremental GC.
+    pub forward: Value,
```

Comment:
Interesting idea. I will try this out. The only critical point is that certain artificial heap blocks, such as `OneWordFiller`, `FreeSpace`, and `FwdPtr` (used by the copying GC) would then have a different offset for the tag than the regular objects with the `Obj` header.

Reply:
Probably not worth it for now. I could well be that the compiler generates just as efficient code for our artificial cost model anyway. Maybe one to consider for the future, after merging the GC.

Reply:
**Update**: With the introduction of the mark bitmap, no linear partition memory scan is now performed by the incremental GC, and one could again consider swapping tag and forwarding pointer. Although, it could be fragile, if we would like to later use linear heap scan at some time.